### PR TITLE
Add video quads

### PR DESCRIPTION
**Pull request by manthey**
Video quads are currently only supported by the canvas renderer.  If the video is not ready, a preview color can be shown, but not a preview image (to show a preview image, create an image quad at the same coordinates as the video quad).

- Added color quads to the canvas renderer.  This made things easier to test.
- Improved some of the caching logic in quads to reduce reloading images and videos in some circumstances.
- Added a utility method to invalidate one or all of the quad's caches.  This removes using a private method in the quad example.
- Added code to defer rendering if something is busy.  This prevents flashing when seeking on a playing video.
- Fixed an issue with the canvas transform on images and videos.
- Update quadFeature documentation.  Also update some renderer documentation.
- Improve testing infrastructure.  Don't test video in PhantomJS.  Allow console.log messages to appear in tutorial test output.
- Added a video to the quads example.
- Add two video quad tutorials; one stand-alone quad and one quad positioned on a map.
- Added tests.

### CODE REVIEW

**Comment by manthey** [comment]
This PR is broken into separate commits to try to make it easier to review.

File: src/canvas/canvasRenderer.js [hunk 1]
```diff
@@ -111,6 +111,13 @@ var canvasRenderer = function (arg) {
         features = layer.features(),
         i;
 
+    for (i = 0; i < features.length; i += 1) {
+      if (features[i]._delayRender()) {
```
Comment (aashish24) [inline src/canvas/canvasRenderer.js]: should it be  if (!features[i]._delayRender())? If delayRender is true which means you want to skip rendering or something I missed here?

Comment (manthey) [reply src/canvas/canvasRenderer.js]: Calling `render()` reschedules the rendering to the next animation frame.  The `return` then stops it from rendering right now.

Comment (aashish24) [reply src/canvas/canvasRenderer.js]: ok, thanks perhaps adding a comment here might help but I am fine either way.

File: src/canvas/quadFeature.js [hunk 2]
```diff
@@ -47,56 +48,157 @@ var canvas_quadFeature = function (arg) {
...
-   * Render all of the image quads.
+   * Render all of the image and video quads.
    *
    * @param {CanvasRenderingContext2D} context2d The rendering context.
    * @param {geo.map} map The current renderer's parent map.
    */
-  this._renderImageQuads = function (context2d, map) {
-    if (!m_quads.imgQuads.length) {
+  this._renderImageAndVideoQuads = function (context2d, map) {
+    if ((!m_quads.imgQuads || !m_quads.imgQuads.length) && (!m_quads.vidQuads || !m_quads.vidQuads.length)) {
```
Comment (aashish24) [inline src/canvas/quadFeature.js]: Can we wrap this long line?

File: src/quadFeature.js [hunk 3]
```diff
@@ -424,31 +451,196 @@ var quadFeature = function (arg) {
...
+      imgQuads: imgQuads,
+      vidQuads: vidQuads,
+      origin: origin
+    };
     return m_quads;
   };
 
   /**
-   * Initialize
+   * If the data has changed and caching has been used, one or all data items
```
Comment (aashish24) [inline src/quadFeature.js]: this sentence seems incomplete or missing some words?

**Pull request merged**